python/tank/bootstrap/manager.py

```diff
@@ -202,6 +203,28 @@ def _set_bundle_cache_fallback_paths(self, paths):
     )
 
 
+    def _get_async_bootstrap(self):
+        """
+        Boolean property that specifies how the engine bootstrap logic should be executed.
+
+        By default, when ```async_bootstrap`` is false, the whole engine bootstrap logic
+        will be executed synchronously in the main application thread.
+
+        When ``async_bootstrap` is true, an :class:`sgtk.Sgtk` instance will be bootstrapped asynchronously
+        in a background thread, followed by launching the engine synchronously
+        in the main application thread. This will allow the main application to continue
```

Comment:
I'm wondering if the engine starting up synchronously is an artificial limitation we're introducing in the API. I see the issue with Qt-based DCCs like maya needing to be bootstrap in the main thread. But I think there are other engines that could bootstrap in asynchronously without trouble. Perhaps we can make it possible and just rely on the documentation outline the potential issues encountered with some engines?